pkg/virt-handler/vm.go

```diff
@@ -645,25 +645,25 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 				var newInterface v1.VirtualMachineInstanceNetworkInterface
 				var isForwardingBindingInterface = false
 
-				if existingInterfacesSpecByName[domainInterface.Alias.Name].Masquerade != nil || existingInterfacesSpecByName[domainInterface.Alias.Name].Slirp != nil {
+				if existingInterfacesSpecByName[domainInterface.Alias.GetName()].Masquerade != nil || existingInterfacesSpecByName[domainInterface.Alias.GetName()].Slirp != nil {
```

Comment:
The tests indicate that  we may have an issue now detecting the interface (the IP is not reported).

Reply:
As of now, I could not find the leakage. With trials, I found followings:

1. If I replace the `name` in Alias type with any variable name starting with a capital letter (i.e. exported/public/visible-outside-of-api), function tests all pass.
2. If I replace the `name` in Alias type with any other package-private/internal  name, the `watch_test` for 'kubectl get vmi` would fail for not showing the IP.

So, I'd like to propose to take this fix for now. I will in the meanwhile work to find out the leakage.

Reply:
Addressed as:
- The issue was because the JSON marshaling/unmarshaling for the communication between the `virt-handler` and `virt-launcher` for the package-private members of `Alias` objects would also require special treatment. This was addressed by adding two functions, `MarshalJSON` and `UnmarshalJSON`, for `Alias` objects.